### DISCUSSION TITLE
dune build failed on Windows.

### DISCUSSION THREAD

**Discussion by CanftIn** [ocaml/dune#7166] [Q&A]
## Expected Behavior
dune build pass.

## Reproduction

- PR with a reproducing test: 

1. By following the [quick-start](https://dune.readthedocs.io/en/stable/quick-start.html#building-a-hello-world-program), make two files: `hello-world.ml`, `dune`. Then build a project named hello-world on Windows. There's a error occurred:
```bash
$ dune build hello_world.exe
Error: exception
Sys_error("C:/cygwin/home/appveyor/.opam/ocaml-variants.4.09.0+mingw64c/lib/ocaml\\Makefile.config:
No such file or directory")
Raised by primitive operation at file "stdlib.ml", line 398, characters 28-54
Called from file "src/stdune\\io.ml", line 55, characters 17-37
Called from file "src/stdune\\io.ml" (inlined), line 107, characters 25-69
Called from file "src/ocaml-config\\ocaml_config.ml", line 312, characters
  16-37
Called from file "src/ocaml-config\\ocaml_config.ml", line 494, characters
  15-45
Called from file "src/dune\\context.ml", line 405, characters 14-36
Called from file "src/fiber\\fiber.ml", line 196, characters 36-41
Called from file "src/fiber\\fiber.ml", line 102, characters 8-15

I must not crash.  Uncertainty is the mind-killer. Exceptions are the
little-death that brings total obliteration.  I will fully express my cases.
Execution will pass over me and through me.  And when it has gone past, I
will unwind the stack along its path.  Where the cases are handled there will
be nothing.  Only I will remain.
```
2. dune path: 
```
D:\linux_home\.opam\4.09.0+mingw64c\bin
```
3. ocaml with cygwin path:
```
D:\OCaml64
```
4. Environment variables `HOME=D:\linux_home`
## Specifications

- Version of `dune` : 2.4.0
- Version of `ocaml` : 4.09.0
- Operating system : Microsoft Windows [Version 10.0.18363.752]

## Additional information 
This error seems to be due to dune can not find cygwin path.
And my cygwin is at `D:\OCaml64\Cygwin.bat`.

**Comment by ghost** [comment]
Does the above file exist on your system?

```
C:\cygwin\home\appveyor\.opam\ocaml-variants.4.09.0+mingw64c\lib\ocaml\Makefile.config
```

**Comment by CanftIn** [comment]
@diml 
This file location on my system is:
```
D:\linux_home\.opam\4.09.0+mingw64c\lib\ocaml\Makefile.config
```
Because I set `HOME` environment variables as `D:\linux_home`.

And my ocaml install path is :
```
D:\OCaml64
```
so dune can not find it.
Is there a way solve this? Thanks.

**Comment by ghost** [comment]
Is this the system you work on or a CI? I'm asking because of the "appveyor" in the path in the error message.

**Comment by CanftIn** [comment]
Oh, the truth is I don't have this 
`C:\cygwin\home\appveyor\.opam\ocaml-variants.4.09.0+mingw64c\lib\ocaml\Makefile.config` 
on my system cause I don't install cygwin in C drive. I'm not familiar with CI, so I get a little confuse that I have no CI system on the Windows may not matter important. 

And dune works fine on my another linux system. 

In `D:\linux_home\.opam\4.09.0+mingw64c\lib\ocaml\Makefile.config`, there is some configs about my local path is
```makefile
## Arguments passed to the configure script

CONFIGURE_ARGS=--prefix=D:/linux_home/.opam/4.09.0+mingw64c --host=x86_64-w64-mingw32 CC=x86_64-w64-mingw32-gcc CPP=x86_64-w64-mingw32-cpp

## Top build directory

TOP_BUILDDIR = .

## Installation directories

prefix=D:/linux_home/.opam/4.09.0+mingw64c
```

I just do a stupid way to create the above path by making fold one by one. And copy `D:\linux_home\.opam\4.09.0+mingw64c\lib\ocaml\Makefile.config` to the created fold.

Then I build dune project. And get another error:
```
λ dune build hello_world.exe
File "command line", line 1:
Error: Unbound module Stdlib
```

BTW, maybe I should focus on linux environment rather than Windows.

**Comment by bschommer** [comment]
Since you did not install cygwin I assume that you have installed a precompiled version of OCaml? In that case you may need to set some environment variables in order for OCaml to find its install path.

**Comment by ghost** [comment]
Indeed, you need to set at least `OCAMLLIB` to point to the standard library directory.
